scripts/hid_configurator/configurator_core.py

```diff
@@ -484,160 +527,99 @@ def perform_dfu(dev, args):
-    sp_devices.required = True
-    for device_name in DEVICE:
-        device_parser = sp_devices.add_parser(device_name)
 
-        sp_commands = device_parser.add_subparsers(dest='command')
-        sp_commands.required = True
+def is_other_dfu_operation(dfu_info):
+    # Check there is no other DFU operation.
+    if dfu_info is None:
+        print('Cannot start DFU, device not responding')
+        return True
 
-        parser_dfu = sp_commands.add_parser('dfu', help='Run DFU')
-        parser_dfu.add_argument('dfu_image', type=str, help='Path to a DFU image')
+    if dfu_info[0] != 0:
+        print('Cannot start DFU, already in progress')
+        return True
 
-        sp_commands.add_parser('fwinfo', help='Obtain information about FW image')
-        sp_commands.add_parser('fwreboot', help='Request FW reboot')
+    return False
 
-        device_config = DEVICE[device_name]['config']
-        if device_config is not None:
-            assert(type(device_config) == dict)
-            parser_config = sp_commands.add_parser('config', help='Configuration option get/set')
 
-            sp_config = parser_config.add_subparsers(dest='module')
-            sp_config.required = True
+def is_ok(dfu_image):
+    if not os.path.isfile(dfu_image):
+        print('DFU image file does not exists')
+        return False
 
-            for module_name in device_config:
-                module_config = device_config[module_name]
-                assert(type(module_config) == dict)
-                module_opts = module_config['options']
-                assert(type(module_opts) == dict)
+    img_length = os.stat(dfu_image).st_size
 
-                parser_config_module = sp_config.add_parser(module_name, help='{} module options'.format(module_name))
-                sp_config_module = parser_config_module.add_subparsers(dest='option')
-                sp_config_module.required = True
+    if img_length <= 0:
+        print('DFU image is empty')
+        return False
 
-                for opt_name in module_opts:
-                    parser_config_module_opt = sp_config_module.add_parser(opt_name, help=module_opts[opt_name].help)
-                    parser_config_module_opt.add_argument('value', type=int, default=None, nargs='?', help='int from range {}'.format(module_opts[opt_name].range))
-    args = parser.parse_args()
+    print('DFU image size: {} bytes'.format(img_length))
 
-    dev = open_device(args.device_type)
-    if not dev:
-        return
+    return True
 
-    if args.command == 'dfu':
-        perform_dfu(dev, args)
-    elif args.command == 'fwinfo':
-        perform_fwinfo(dev, args)
-    elif args.command == 'fwreboot':
-        perform_fwreboot(dev, args)
-    elif args.command == 'config':
-        perform_config(dev, args)
 
 if __name__ == '__main__':
     try:
-        configurator()
+        print("Please run configurator_cli.py or gui.py to start application")
     except KeyboardInterrupt:
```

Comment:
I think try except here is not necessary. We are just making one print so we should not expect KeyboardInterrupt

Reply:
Done.